engine/controllers/models.cc

```diff
@@ -192,4 +192,56 @@ void Models::DeleteModel(const HttpRequestPtr& req,
     resp->setStatusCode(k404NotFound);
     callback(resp);
   }
+}
+
+void Models::SetModelAlias(
+    const HttpRequestPtr& req,
+    std::function<void(const HttpResponsePtr&)>&& callback) const {
+  if (!http_util::HasFieldInReq(req, callback, "modelId") ||
+      !http_util::HasFieldInReq(req, callback, "modelAlias")) {
+    return;
+  }
+  auto model_handle = (*(req->getJsonObject())).get("modelId", "").asString();
+  auto model_alias = (*(req->getJsonObject())).get("modelAlias", "").asString();
+  LOG_DEBUG << "GetModel, Model handle: " << model_handle
+            << ", Model alias: " << model_alias;
+
+  modellist_utils::ModelListUtils modellist_handler;
+  try {
+    if (modellist_handler.UpdateModelAlias(model_handle, model_alias)) {
+      std::string message = "Successfully set model alias '" + model_alias +
+                            "' for modeID '" + model_handle + "'.";
+      LOG_INFO << message;
+      Json::Value ret;
+      ret["result"] = "OK";
+      ret["modelHandle"] = model_handle;
+      ret["message"] = message;
+      auto resp = cortex_utils::CreateCortexHttpJsonResponse(ret);
+      resp->setStatusCode(k200OK);
+      callback(resp);
+    } else {
+      std::string message = "Unable to set model alias for modelID '" +
+                            model_handle + "': model alias '" + model_alias +
+                            "' or modelID is not unique!";
```

Comment:
Nitpick: This scenario would only occur if model alias is already used right? 

- User might be confused by an error message suggesting `model_id` is not unique
- As an aside, how do we plan to deal with non-unique `model_id`s?

Reply:
- Yes, let me modify the error message, this scenario occurs when model alias is already in use or equal to model_id of other models.
- I think the model-id should be unique to make model entry can be distinguished, alias just a shorter name for easier to use.